src/core/sys/linux/sys/time.d

```diff
@@ -0,0 +1,72 @@
+/**
+ * D header file for POSIX.
+ *
+ * Copyright: Copyright Sociomantic Labs GmbH.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Leandro Lucarella
+ */
+
+/*          Copyright Sociomantic Labs GmbH.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module core.sys.linux.sys.time;
+
+private import core.sys.linux.config;
+public import core.sys.posix.sys.types;  // timeval
+
+version (linux)
+else static assert(false, "Unsupported platform");
+
+version( CRuntime_Glibc )
+else static assert(false, "Unsupported platform");
```

Comment:
Well, you could actually just version the whole core.sys.linux.time file like I said (with the two versions required), and then if someone imports the files and tries to use the functions they won't be defined. Look for instance at: https://github.com/dlang/druntime/blob/master/src/core/sys/linux/sys/auxv.d at the top. The precedent is pretty clear.

I do agree that people aren't going to have thought of the difference between glibc and non glibc. Most people just say version(linux) and call it a day.

However, in this case, you are doing macros. Glibc isn't really required. I'm not sure we need it.